scipy/linalg/tests/test_basic.py

```diff
@@ -644,12 +654,14 @@ def test_random_sym_complex(self):
             x = solve(a, b, assume_a="pos")
             assert_array_almost_equal(dot(a, x), b)
 
-    def test_check_finite(self):
-        a = [[1, 20], [-30, 4]]
+    @array_api_compatible
+    def test_check_finite(self, xp):
+        a = xp.asarray([[1, 20], [-30, 4]], dtype=xp.float64)
         for b in ([[1, 0], [0, 1]], [1, 0],
                   [[2, 1], [-30, 4]]):
+            b = xp.asarray(b, dtype=xp.float64)
```

Comment:
Integer inputs are here for testing the conversion to the relevant float types in the functions. We should not modify them.